TypeScript.YAML-tmLanguage

```diff
@@ -76,10 +76,21 @@ repository:
 
   object-heritage:
     name: meta.object.heritage.ts
-    match: (?:\b(extends|implements)\b|,)(?:\s+([a-zA-Z_$][.\w$]*))
+    begin: '(?:\s*\b(extends|implements))'
+    beginCaptures:
+      '1': { name: keyword.other.ts }
+    end: (?=\{)
+    endCaptures:
+      '1': { name: brace.curly.ts }
+    patterns:
+    - include: '#object-heritage-parent'
+    - include: '#comment'
+  
+  object-heritage-parent:
+    name: meta.object.heritage.parent.ts
+    match: '(?:\s*([a-zA-Z_$][\w$]*))'
     captures:
-       '1': { name: keyword.other.ts }
-       '2': { name: storage.type.ts } 
+      '1': { name: storage.type.ts }
```

Comment:
Does this cover the following case (from [ts spec](https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md)):
```
interface KeyValuePair<K, V> extends Array<K | V> { 0: K; 1: V; }
```